### PR TITLE
Vector*: Added random()

### LINKED ISSUE #18996: request Vector.randomUnitVector()
Hey, 
I've seen this in a handful of other libraries and always wondered why it wasn't included. If you don't use random vectors, then what are you doing with you life? I use it so often I add it in almost all my projects, it looks like this;
```
THREE.Vector3.prototype.randomUnitVector = function(){
    this.x = Math.random() * 2 - 1;
    this.y = Math.random() * 2 - 1;
    this.z = Math.random() * 2 - 1;
    this.normalize();
    return this;
}
```
You would call it like this;
`let v = new THREE.Vector3().randomUnitVector();`

The name is quite long, maybe in relations to other vector functions names like .randomize() (like normalize) or just .random() would be more logical? 
You could also give it an argument to set the length, but I find it more logical to call .setLength() afterwards because it wouldn't be a unit vector anymore and it's easier to read.
I'm curious of what you think of it.

Kind regards, Ethan.

### PR DESCRIPTION
**Pull request by EthanHermsey**
Fixed #18996.

Adding a method that sets the components of the vector with a random
number [0-1].

### CODE REVIEW

File: src/math/Vector2.d.ts [hunk 1]
```diff
@@ -457,6 +457,11 @@ export class Vector2 implements Vector {
 	 *
 	 * @see {@link http://en.wikipedia.org/wiki/Taxicab_geometry|Wikipedia: Taxicab Geometry}
 	 */
-	manhattanDistanceTo( v: Vector2 ): number;
+    manhattanDistanceTo( v: Vector2 ): number;
```
Comment (Mugen87) [inline src/math/Vector2.d.ts:460]: Please use hard instead of soft tabs.

**Comment by Mugen87** [comment]
It would be great if you can also update the docs 👍 


https://threejs.org/docs/index.html#api/en/math/Vector2
https://threejs.org/docs/index.html#api/zh/math/Vector2 (just copy over the English version)
https://threejs.org/docs/index.html#api/en/math/Vector3
https://threejs.org/docs/index.html#api/zh/math/Vector3
https://threejs.org/docs/index.html#api/en/math/Vector4
https://threejs.org/docs/index.html#api/zh/math/Vector4

**Comment by EthanHermsey** [comment]
I have added the hard tabs and changed the docs. Is it done correctly?
I am not certain about the like 'Don't include build files in your Pull Request' on your article about contributing. Those files are forked, do it remove them or do you mean don't change anything inside the build files? 

Also, in the documentation of all vectors, the use of 'the components of THIS vector' and 'the components of THE vector' are used interchangeably. Just wanted to point that out.

**Comment by Mugen87** [comment]
> Those files are forked, do it remove them or do you mean don't change anything inside the build files?

The idea is not to include files from the `build` directory in PRs. You are not doing this so the PR looks fine.

> Also, in the documentation of all vectors, the use of 'the components of THIS vector' and 'the components of THE vector' are used interchangeably. Maybe it's nice to use only one of them?

Yes, improve the wording is a good idea. But please in a separate PR.

File: src/math/Vector2.js [hunk 2]
```diff
@@ -482,6 +482,15 @@ Object.assign( Vector2.prototype, {
 
 		return this;
 
+	},
+
```
Comment (Mugen87) [inline src/math/Vector2.js:486]: Please remove all trailing spaces from your PR.

BTW: If you work with editors like VSCode, this is done automatically by the eslint autocorrect.

Comment (EthanHermsey) [reply src/math/Vector2.js]: Hhmm I tried this with the default eslint settings. It does remove the trails but messes up the rest of the document. For now I did it by hand, I will have to figure this out, using the eslint-config-mdcs. It's very nice.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**